velox/functions/prestosql/aggregates/tests/MinMaxByAggregationTest.cpp

```diff
@@ -13,273 +13,988 @@
+      case TypeKind::BIGINT:                                      \
+        EXECUTE_TEST_BY_VALUE_TYPE(testFunc, int64_t, isPartial); \
+        break;                                                    \
+      case TypeKind::REAL:                                        \
+        EXECUTE_TEST_BY_VALUE_TYPE(testFunc, float, isPartial);   \
+        break;                                                    \
+      case TypeKind::DOUBLE:                                      \
+        EXECUTE_TEST_BY_VALUE_TYPE(testFunc, double, isPartial);  \
+        break;                                                    \
+      default:                                                    \
+        LOG(FATAL) << "Unsupported value type of minmax_by(): "   \
+                   << mapTypeKindToName(GetParam().valueType);    \
+    }                                                             \
+  } while (0);
+
+std::string rowVectorToString(const RowVectorPtr& rowVector) {
+  std::string str;
+  for (size_t i = 0; i < rowVector->childrenSize(); ++i) {
+    str += fmt::format("Column C{}:\n", i);
+    auto child = rowVector->childAt(i);
+    // If verbose mode, print the whole vector.
+    for (size_t j = 0; j < child->size(); ++j) {
+      str += fmt::format("\tC{}[{}]:{}\n", i, j, child->toString(j));
+    }
+  }
+  return str;
+}
+
+class MinMaxByAggregationTestBase : public AggregationTestBase {
+ protected:
+  MinMaxByAggregationTestBase() : numValues_(6) {}
+
+  void SetUp() override {
+    for (const TypeKind type : kSupportedMinmaxByTypeSet) {
+      switch (type) {
+        case TypeKind::TINYINT:
+          dataVectorsByType_.emplace(type, buildDataVector<int8_t>());
+          break;
+        case TypeKind::SMALLINT:
+          dataVectorsByType_.emplace(type, buildDataVector<int16_t>());
+          break;
+        case TypeKind::INTEGER:
+          dataVectorsByType_.emplace(type, buildDataVector<int32_t>());
+          break;
+        case TypeKind::BIGINT:
+          dataVectorsByType_.emplace(type, buildDataVector<int64_t>());
+          break;
+        case TypeKind::REAL:
+          dataVectorsByType_.emplace(type, buildDataVector<float>());
+          break;
+        case TypeKind::DOUBLE:
+          dataVectorsByType_.emplace(type, buildDataVector<double>());
+          break;
+        default:
+          LOG(FATAL) << "Unsupported data type: " << mapTypeKindToName(type);
+      }
+    }
+    ASSERT_EQ(dataVectorsByType_.size(), kSupportedMinmaxByTypeSet.size());
+    rowVectors_ = makeVectors(rowType_, 10, 100);
+    createDuckDbTable(rowVectors_);
+  }
+
+  // Build a flat vector with numeric native type of T. The value in the
+  // returned flat vector is in ascending order.
+  template <typename T>
+  FlatVectorPtr<T> buildDataVector() {
+    std::vector<T> values;
+    values.reserve(numValues_);
+    for (int i = 1; i <= numValues_; ++i) {
+      values.push_back(i);
+    }
+    return makeFlatVector<T>(values);
```

Comment:
nit: This can be shortened to `makeFlatVector<T>(numValues_, [] (auto row) {return row;})`

Reply:
Nice. Thanks for the suggestion!